### DISCUSSION TITLE
What is the meaning of `type` keyword?

### DISCUSSION THREAD

**Discussion by brsanthu** [eclipse-langium/langium#1476] [Q&A] [answered]
I have seen this usage in the SQL sample language. `type` is not documented in the grammar. Can somebody elaborate what is this used for?

```
type TableVariableSource = TableSourceItem | SubQuerySourceItem;
```

**Comment by msujew** [comment, marked as answer]
Hey @brsanthu,

this feature of Langium is outlined in [here](https://langium.org/docs/sematic-model/#declared-types). It's not really part of the grammar language so it has its own documentation.

**Reply by bogdanbiv** [reply]
As the link provided here leads to a 404 like doc, I assume this concept of types is deprecated?

**Reply by msujew** [reply]
No, we just restructured our documentation. See [here](/docs/reference/semantic-model/#declared-types) for an updated link.

**Reply by brsanthu** [reply]
@msujew your link is erroring again. I think this is the new link. https://langium.org/docs/reference/semantic-model/#declared-types
